app/Components/Sudoku Board/Functions/BoardFunctions.ts

```diff
@@ -186,159 +61,48 @@ export const formatTime = (inputSeconds: number) => {
-  }
-  return solArray;
-}
-
-export function getNextHint(board: any, solution: any, strategies: any) {
-  let hintStrategies = [...strategies];
-  hintStrategies.push("AMEND_NOTES", "SIMPLIFY_NOTES");
-  let boardArray = componentBoardValsToArray(board);
-  let notesArray = componentBoardNotesToArray(board);
-  let solutionArray = componentSolutionValsToArray(solution);
-  let hint;
-  try {
-    hint = getHint(boardArray, notesArray, hintStrategies, solutionArray);
-  } catch (e) {
-    console.log(e);
-  }
-  return hint;
-}
-
-export function parseApiAndAddNotes(
-  board: any,
-  puzzleCurrentNotesState: any,
-  isDrill: boolean
-) {
-  if (!puzzleCurrentNotesState) {
-    return;
-  }
-  if (puzzleCurrentNotesState.length != 729) {
-    return;
-  }
-  let stringIndex = 0;
-  for (let i = 0; i < 9; i++) {
-    for (let j = 0; j < 9; j++) {
-      for (let currNoteIndex = 0; currNoteIndex < 9; currNoteIndex++) {
-        stringIndex = 81 * i + 9 * j + currNoteIndex;
-
-        if (puzzleCurrentNotesState.charAt(stringIndex) == 1) {
-          if (isDrill) {
-            board = addNumberAsNote(currNoteIndex + 1, board, j, i);
-          } else {
-            board = addNumberAsNote(currNoteIndex + 1, board, i, j);
-          }
-        }
-      }
-    }
-  }
-  return board;
-}
-
-export function strPuzzleToArray(str: any) {
-  let arr: any = [];
-  for (let i = 0; i < str.length; i += 9) {
-    arr.push(
-      str
-        .slice(i, i + 9)
-        .split("")
-        .map(Number)
-    );
-  }
-  // @ts-ignore
-  let output = arr[0].map((_: any, colIndex: any) =>
-    arr.map((row) => row[colIndex])
-  );
-  return { puzzle: output };
-}
-
-export async function saveGame(activeGame: any, timer: any) {
-  activeGame.currentTime = timer;
-
+export async function saveGame(activeGame: SudokuObjectProps) {
   Puzzles.saveGame(activeGame).then((res: any) => {
-    if (activeGame.numWrongCellsPlayed == null) {
-      activeGame.numWrongCellsPlayed = 0;
-    }
     if (res) {
       console.log("Game progress was saved successfully!");
     }
   });
 }
 
-export async function finishGame(showResults: any) {
-  Puzzles.finishGame().then((res: gameResults) => {
-    if (res) {
-      showResults(
-        res.score,
-        res.solveTime,
-        res.numHintsUsed,
-        res.numWrongCellsPlayed,
-        res.difficulty
-      );
-    }
-  });
+/**
+ * Calculates and returns the score of the game.
+ * Calls Puzzle.finishGame to update Statistics Page and remove localstorage object.
+ * @param difficulty
+ * @param numHintsUsed
+ * @param numWrongCellsPlayed
+ * @param time
+ * @returns
+ */
+export function finishGame(
+  difficulty: string,
+  numHintsUsed: number,
+  numWrongCellsPlayed: number,
+  time: number
+): number {
+  const numericalDifficulty: GameDifficultyScore =
+    difficulty === "easy"
+      ? 10
+      : difficulty === "medium"
+      ? 20
+      : difficulty === "hard"
+      ? 30
+      : 10;
+  // calculate score
```

Comment:
nit: the score calculation uses a lot of ternary operators and doesn't have any comments making it hard to read. Also I can't tell if numericalDifficulty / 1000 will be equal to 0 or 0.01 for 10 cause of float vs int. Would be nice to have some comments explaining how this all works and maybe an example score calculation

Reply:
Good call, I'll work on docs/comments for that and clean up that code a bit